bundles/org.eclipse.text/src/org/eclipse/jface/text/source/AnnotationModel.java

```diff
@@ -653,11 +655,10 @@ private void cleanup(boolean fireModelChanged, boolean forkNotification) {
 				}
 			} else {
 				synchronized (mapLock) {
-					annotations.forEach((a, p) -> {
-						if (p == null || p.isDeleted()) {
-							deleted.add(a);
-						}
-					});
+					deleted = annotations.entrySet().parallelStream()//
+							.filter(e -> e.getValue() == null || e.getValue().isDeleted)//
```

Comment:
`e.getValue` might be extracted into a local variable to prevent double method call.